prov/rxm/src/rxm_av.c

```diff
@@ -83,14 +79,21 @@ rxm_av_insert_cmap(struct fid_av *av_fid, const void *addr, size_t count,
 				       ofi_av_lookup_fi_addr(av, cur_addr));
 			if (fi_addr_tmp == FI_ADDR_NOTAVAIL)
 				continue;
+
+			if (rxm_ep->util_ep.domain->data_progress == FI_PROGRESS_MANUAL)
+				ofi_ep_lock_acquire(&rxm_ep->util_ep);
```

Comment:
cmap->lock would be NULL in case of manual progress. So we need av lock and ep lock to serialize access to av->ep_list and cmap (an EP resource) respectively.

Reply:
If you're protecting access to the cmap, then we should either always use the cmap lock or the ep lock.  Don't change locks based on progress.  It sounds like the issue here was setting cmap lock to null, when it shouldn't be.

Reply:
ep->lock can be NULL depending on threading level requested by the app. That leaves us with having cmap->lock available always which means in case of manual progress, we are incurring additional cost of acquiring and releasing locks unnecessarily.